### ISSUE TITLE
Server.path + F5 on login page

### ISSUE THREAD

**Issue by rokiden** [authelia/authelia#5902]
### Version

v4.37.5

### Deployment Method

Docker

### Reverse Proxy

Traefik

### Description

0. Traefik and Authelia configured to subpath `auth`
1. Client was redirected from secured page to Authelia login page: `example.com/auth/?rd=...`
2. Authelia login page JavaScript drops trailing slash after subpath: `example.com/auth/?rd=... -> example.com/auth?rd=...`
3. User refresh page, Authelia server drops subpath with redirect 301 to / : `example.com/auth?rd=... -> 301 example.com/?rd=...`
4. Traefik returns 404 to non-subpath request.

After experiments with pure Authelia i got this:
Request -> Response Code
`localhost:9091/` -> 200
`localhost:9091/auth` -> 200
`localhost:9091/auth/` -> 200 and js drops slash
`localhost:9091/auth/?rd=a` -> 200 and js drops slash
`localhost:9091/auth?rd=a` -> 301 `localhost:9091/?rd=a`

I can't understand why server response 301 after valid subpath request.

### Reproduction

[yml.zip](https://github.com/authelia/authelia/files/12432573/yml.zip)
```
mkdir -p /tmp/authelia/config && cd /tmp/authelia
unzip ~/Downloads/yml.zip -d config
podman run --rm -d -v ./config:/config -p 9091:9091 authelia/authelia
curl -i localhost:9091/auth
# 200 OK
curl -i localhost:9091/auth?
# 301 Moved Permanently http://localhost:9091/
```

### Expectations

`localhost:9091/auth?rd=a` -> 200 OK
OR 
JS doesn't drop slash

### Configuration (Authelia)

```yaml
---
theme: auto
jwt_secret: c0849e6890b9d711013e6c0412ba7d2b47e9de92c8781b52f39801e98382bdc5
default_2fa_method: ""
server:
  host: 0.0.0.0
  port: 9091
  path: "auth"
  enable_pprof: false
  enable_expvars: false
  disable_healthcheck: false
  headers:
    csp_template: ""
log:
  level: trace
telemetry:
  metrics:
    enabled: false
    address: tcp://0.0.0.0:9959
totp:
  disable: true
  issuer: authelia.com
  algorithm: sha1
  digits: 6
  period: 30
  skew: 1
  secret_size: 32
webauthn:
  disable: false
  timeout: 60s
  display_name: Authelia
  attestation_conveyance_preference: indirect
  user_verification: preferred
ntp:
  address: "time.cloudflare.com:123"
  version: 4
  max_desync: 3s
  disable_startup_check: false
  disable_failure: false
authentication_backend:
  password_reset:
    disable: false
    custom_url: ""
  refresh_interval: 5m
  file:
    path: /config/users.yml
    password:
      algorithm: argon2
      argon2:
        variant: argon2id
password_policy:
  standard:
    enabled: false
    min_length: 8
    max_length: 0
    require_uppercase: true
    require_lowercase: true
    require_number: true
    require_special: true
  zxcvbn:
    enabled: false
    min_score: 3
access_control:
  default_policy: one_factor
session:
  name: authelia_session
  domain: example.com
  same_site: lax
  secret: insecure_session_secret
  expiration: 1h
  inactivity: 5m
  remember_me_duration: 1M
regulation:
  max_retries: 3
  find_time: 2m
  ban_time: 5m
storage:
  encryption_key: 734c312b3b7255f40efcd4b4de88afbc
  local:
    path: /config/db.sqlite3
notifier:
  disable_startup_check: false
  filesystem:
    filename: /config/notification.txt
...
```

### Logs (Authelia)

```shell
time="2023-08-24T18:59:51Z" level=warning msg="Configuration: access control: no rules have been specified so the 'default_policy' of 'one_factor' is going to be applied to all requests"
time="2023-08-24T18:59:51Z" level=info msg="Authelia v4.37.5 is starting"
time="2023-08-24T18:59:51Z" level=info msg="Log severity set to trace"
time="2023-08-24T18:59:51Z" level=trace msg="Starting scan of directory  for certificates"
time="2023-08-24T18:59:51Z" level=trace msg="Finished scan of directory  for certificates"
time="2023-08-24T18:59:51Z" level=info msg="Storage schema is being checked for updates"
time="2023-08-24T18:59:51Z" level=info msg="Storage schema is already up to date"
time="2023-08-24T18:59:51Z" level=debug msg="The NTP startup check was skipped due to there being no configured 2FA access control rules"
time="2023-08-24T18:59:51Z" level=info msg="Initializing server for non-TLS connections on '[::]:9091' path '/' and '/auth'"
time="2023-08-24T18:59:59Z" level=trace msg="Request hit" method=GET path=/ remote_ip=10.0.2.100
time="2023-08-24T18:59:59Z" level=trace msg="Replied (status=200)" method=GET path=/ remote_ip=10.0.2.100
time="2023-08-24T19:00:02Z" level=trace msg="Request hit" method=GET path=/ remote_ip=10.0.2.100
time="2023-08-24T19:00:02Z" level=trace msg="Replied (status=301)" method=GET path=/ remote_ip=10.0.2.100
time="2023-08-24T19:00:22Z" level=trace msg="Request hit" method=GET path=/api/health remote_ip="::1"
time="2023-08-24T19:00:22Z" level=trace msg="Replied (status=200)" method=GET path=/api/health remote_ip="::1"
```

### Logs (Proxy / Application)

```shell
[user@pc tmp]$ curl -i localhost:9091/auth
HTTP/1.1 200 OK
Date: Thu, 24 Aug 2023 18:59:59 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 993
X-Content-Type-Options: nosniff
Referrer-Policy: strict-origin-when-cross-origin
Permissions-Policy: interest-cohort=()
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 1; mode=block
Content-Security-Policy: default-src 'self'; frame-src 'none'; object-src 'none'; style-src 'self' 'nonce-Ma1zyKuU3Grlu7omEHnbrO0E2iBOS14S'; frame-ancestors 'none'; base-uri 'self'

<!DOCTYPE html>
<html lang="en">
<head>
  <base href="http://localhost:9091/auth/" />
  <meta property="csp-nonce" content="Ma1zyKuU3Grlu7omEHnbrO0E2iBOS14S" />
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="theme-color" content="#000000" />
  <meta name="description" content="Authelia login portal for your apps" />
  <link rel="manifest" href="./manifest.json" />
  <link rel="icon" href="./favicon.ico" />
  <title>Login - Authelia</title>
  <script type="module" crossorigin src="./static/js/index.fad9e36b.js"></script>
  <link rel="stylesheet" href="./static/css/index.40feef90.css">
</head>

<body
    data-basepath="/auth"
    data-duoselfenrollment="false"
    data-logooverride="false"
    data-rememberme="true"
    data-resetpassword="true"
    data-resetpasswordcustomurl=""
    data-theme="auto"
>
  <noscript>You need to enable JavaScript to run this app.</noscript>
  <div id="root"></div>
  
</body>
</html>
[user@pc tmp]$ curl -i localhost:9091/auth?
HTTP/1.1 301 Moved Permanently
Date: Thu, 24 Aug 2023 19:00:01 GMT
Content-Length: 0
Location: http://localhost:9091/
```

### Documentation

https://www.authelia.com/configuration/miscellaneous/server/#path

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] This is a bug report and not a support request

- [X] I have read the security policy and this bug report is not a security issue or security related issue

- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration

- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide

- [X] I have checked for related proxy or application logs and included them if available

- [X] I have checked for related issues and checked the documentation

**Comment by rokiden** [comment]
Some debugging results:
* master d241a371dc08eac07123b5e41eff3751da56e69d behaves the same
* redirect 301 comes from [fasthttp.router](https://github.com/fasthttp/router/blob/b2b7c5e96963bfa3201e902c94783e9ab0baf90b/router.go#L379) and can be disabled by router option RedirectFixedPath=false, my case fixed, but not sure it's totally safe

**Comment by james-d-elliott** [comment]
Yeah I am not 100% sure I'm correct, and I've not specifically validated it as an issue yet, but I think the issue is specifically in the react router. The path should be `https://localhost:9091/auth/` per standard web semantics. It shouldn't redirect you like it presumably is.

**Comment by james-d-elliott** [comment]
I can't seem to reproduce this: https://auth.jameselliott.dev/authelia

**Comment by james-d-elliott** [comment]
Looks like this was fixed by react-router and has therefore been fixed in 4.38.0, or at least it's working for me.

**Comment by james-d-elliott** [comment]
Never mind.. still an issue.
